### PR TITLE
refactor: strict typechecks

### PR DESCRIPTION
**Pull request by pi0**
enable strict type checks across repo on 2.x branch for safer bw/fw compatibility.

There might be slight type breaking changes.

### CODE REVIEW

File: src/build.ts [hunk 1]
```diff
@@ -499,17 +495,17 @@ async function _build(nitro: Nitro, rollupConfig: RollupConfig) {
   const rewriteRelativePaths = (input: string) => {
     return input.replace(/([\s:])\.\/(\S*)/g, `$1${rOutput}/$2`);
   };
-  if (buildInfo.commands.preview) {
+  if (buildInfo.commands!.preview) {
```
Comment (danielroe) [inline src/build.ts:498]: It feels like these non null assertions might be indicative of a need to improve a type somewhere.

Comment (pi0) [reply src/build.ts:498]: Well it is already set in here when constructing object https://github.com/unjs/nitro/blob/db5f87e5ccb4c7e092f48094fcef8902e781f311/src/build.ts#L471 

`NitroBuildInfo` (the public interface) is intentionally partial to make sure external consumers take this into account.

Comment (danielroe) [reply src/build.ts:498]: I understand. But I’m suggesting that we should have a validator function or internal type that is more accurate than that one.

**Pull request merged**